model/dependencies.go

```diff
@@ -6,33 +6,34 @@ import (
 )
 
 type dependencyIncluder struct {
-	Project   *Project
-	requester string
-	included  map[TVPair]bool
+	Project                         *Project
+	requester                       string
+	included                        map[TVPair]bool
+	deactivateGeneratedVariantTasks map[string]bool
```

Comment:
One further problem is that a task or variant having a "specific activation" at this point actually means that it is meant to be inactive ([here](https://github.com/evergreen-ci/evergreen/blob/main/model/lifecycle.go#L1157) and [here](https://github.com/evergreen-ci/evergreen/blob/main/model/lifecycle.go#L1565) among others). My best guess is that the name meant that we are "specifically" setting a task/variant to `active: false` hence its name.  

`deactivateGeneratedVariantTasks` represents if a dependency should be added to the activation info's list of tasks (i.e. list to deactivate).  It's a `map[string]bool` because we only want to deactivate a generated dependency everything that depends on it is also marked inactive. If even one task that depends on the generated dependency is active, we'll need to activate the generated dependency as well.  If we encounter this case then the value in the map is set to `false` and we skip it later when adding tasks to the activation list.

Perhaps a name like `deactivateGeneratedDep` is more accurate, I'll rename for now but we can discuss.

Reply:
Got it. I think I was confused because `activate` is really exposing an option to deactivate, so `activationVariants` are those that explicitly set `active: false`. Thanks for the clear explanation!